release/models/policy-forwarding/openconfig-pf-forwarding-policies.yang

```diff
@@ -264,14 +270,72 @@ submodule openconfig-pf-forwarding-policies {
         according to the IP destination).";
     }
 
+    leaf decap-network-instance {
+      type leafref {
+        // this must be an absolute reference to allow another NI
+        // to be referenced.
+        path "/network-instances/network-instance/config/name";
+      }
+      description
+        "This leaf is mutually exclusive with ../network-instance.
+        This leaf is normally used with NIs (network instances) that are
+        populated by SDN controllers (e.g. via gRIBI), where we want:
+          - The decapsulation decision to be controlled by the
+            controllers.
+          - The new header after decapsulation is to be looked up in
+            another NI, and the NI is chosen by fields (e.g. source IP,
+            DSCP, etc) that are not used for route resolution.
+        When this leaf is set, packets matching the match criteria for
+        the forwarding rule should be looked up in the specified NI.
+        It is expected that the NI should only contain routes that have
+        next hop action as decapsulation.
+          - If there is a match, the packet should be decapsulated.
+            The route lookup on the new header should happen in the
+            ../post-decap-network-instance.
+          - If there is no match, the packet should be looked up in the
+            NI set by ../decap-fallback-network-instance.";
```

Comment:
How is this leaf different than a leaf network-instance which simply contains match criteria which all have decap action?  In other words, do we need a decap-network-instance leaf at all?  Could we use some network-instance populated exclusively with match criteria with decap action set?  The network-instance name could include the substring 'decap' as a hint to it's purpose.

Reply:
Is the difference that  `post-decap-network-instance` is expected to be used together / chained  with `decap-network-instance`?  If so, perhaps it will help to also add this expectation into the description for `decap-network-instance`.

Reply:
Updated the description. Thanks!

Regarding why we don't "reuse" the existing "network-instance", we think explicitly setting a leave is more reliable than string parsing the VRF name. it gives an explicit signal which would make implementation robust.

Reply:
Updated the description. Thanks!

Regarding why we don't "reuse" the existing "network-instance", we think explicitly setting a leave is more reliable than string parsing the VRF name. it gives an explicit signal which would make implementation robust.

Reply:
SGTM